### PR TITLE
Bring back to backdropimage.

### PR DESCRIPTION
**Pull request by ghost**
Follow-up to #6

### CODE REVIEW

**Comment by ghost** [comment]
Made a form alter to backdropimage dialog, to bring back the "Browse server" button.
Note that I choose to make images referenced as "External", to avoid conflicts about file usage...

File: imce.module [hunk 1]
```diff
@@ -210,4 +210,29 @@ function imce_user_page_access($account, $user = FALSE) {
  */
 function imce_reg_dir($dirname) {
   return $dirname == '.' || is_int($dirname) || (is_string($dirname) && $dirname != '' && !preg_match('@(^\s)|(^/)|(^\./)|(\s$)|(/$)|(/\.$)|(\.\.)|(//)|(\\\\)|(/\./)@', $dirname));
+}
+
+function imce_form_alter(&$form, &$form_state, $form_id) {
```
Comment (docwilmot) [inline imce.module]: missing docblock

File: imce.module [hunk 2]
```diff
@@ -210,4 +210,29 @@ function imce_user_page_access($account, $user = FALSE) {
...
+  if ($form_id !== 'filter_format_editor_image_form') {
```
Comment (docwilmot) [inline imce.module]: could we use hook_form_FORM_ID-alter?

File: imce.module [hunk 3]
```diff
@@ -210,4 +210,29 @@ function imce_user_page_access($account, $user = FALSE) {
...
+  }
+  $attributes['onclick'] = 'javascript:'.$openFileBrowser.'return false;';
+
+  $form['image']['browse'] = array(
+    '#type' => 'item',
+    '#markup' => l(t('Browse server'), '#', array('attributes' => $attributes)),
+    '#parents' => array('attributes', 'src'),
+    '#weight' => 0,
+  );
 }
```
Comment (docwilmot) [inline imce.module]: newline missing

**Comment by docwilmot** [comment]
Hi @gifad, wont have time to test this this morning, but I made some code comments and if you could fix those, you could merge the PR once you're comfortable that it works OK.

I am just the person who ported this, because I needed it, but I know little about this module and would be happy if you wanted to join as maintainer.

**Comment by ghost** [comment]
> but I know little about this

Me too, plus I'm a php beginner, plus I could not yet figure out how this kind of hybrid plugin work
This is just a hack

> because I needed it

Note that I registered IMCE images as "external" (via Image URL, not Image Upload), because I don't trust (because I don't understand) the way IMCE deals with managed files, file usage, etc...

**Comment by ghost** [comment]
Comments fixed.
I made a limited extent of tests, but I used this JS command line for years, and it just fills the URL textfield, nothing more, so I think you can trust it...

The way I manage the toggle of the pseudo tabs is very basic, but I did not want to interfere with the tricky JS that handle this
BTW it's flawed : the initial visibility is not assigned to the right "tab" !
This is a minor issue to me : I think IMCE is for people who know what they are doing, not the casual contributor...

**Comment by ghost** [comment]
> BTW it's flawed : 

No, I was wrong, it's just very susceptible...
Made my patch even more basic : the button is now just a `#suffix` of the textfield, instead of a separate (and offending) field (and yes, it's a real button).

**Comment by vtad** [comment]
Works fine, please make a new release.

**Comment by docwilmot** [comment]
Merging this without testing !!! Been reviewed by two people so I guess its working well enough.

**Pull request merged**